### ISSUE TITLE
Is the padding token actually applied?

### ISSUE THREAD

**Issue by hogru** [MolecularAI/pysmilesutils#4]
The Tokenizer is initialized with `padding_token: str = " "`, but it appears that it's actually not used for padding? The encoded SMILES is not padded and by looking at the source code I could not find a function to pad the output before returning it. However, it's added to the vocabulary. At least, that's my impression, I have not studied the code in detail.

**Comment by EBjerrum** [comment]
Yes, the padding token is assigned an index of 0. This makes it compatible with code for padding e.g. https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pad_sequence.html

We don't pre-pad sequences, as we often want to miniminze padding (see e.g. the BucketBatchSampler, which can significantly speed up training of transformers by buncling similar lenght sequences for more effiecent dynamically pad).
